cmd/backend/main.go

```diff
@@ -4,57 +4,74 @@ import (
 	"encoding/json"
 	"flag"
 	"fmt"
-	"github.com/inovex/godays-demo/pkg"
+	"github.com/opentracing/opentracing-go"
 	"log"
 	"net/http"
 	"time"
+
+	"github.com/inovex/godays-demo/pkg"
 )
 
 var port int
+var propagator = pkg.InitHttpHeaderPropagator()
 
 func init() {
 	flag.IntVar(&port, "port", 8080, "port to listen on")
 }
 
 var toasts = []pkg.Toast{
 	{
-		Name: "Hawaii",
+		Name:    "Hawaii",
 		Weekday: time.Monday,
 	},
 	{
-		Name: "Peperoni",
+		Name:    "Peperoni",
 		Weekday: time.Tuesday,
 	},
 	{
-		Name: "Cheese",
+		Name:    "Cheese",
 		Weekday: time.Wednesday,
 	},
 	{
-		Name: "Ham",
+		Name:    "Ham",
 		Weekday: time.Thursday,
 	},
 	{
-		Name: "Caprese",
+		Name:    "Caprese",
 		Weekday: time.Friday,
 	},
 	{
-		Name: "Avocado",
+		Name:    "Avocado",
 		Weekday: time.Saturday,
 	},
 	{
-		Name: "Honey",
+		Name:    "Honey",
 		Weekday: time.Sunday,
 	},
 }
 
-
 func toastsHandler(w http.ResponseWriter, r *http.Request) {
+	var span opentracing.Span
+	context, err := propagator.Extract(opentracing.HTTPHeadersCarrier(r.Header))
+	if err != nil {
+		log.Printf("Could not extract SpanContext from request %s", err)
+		span = opentracing.GlobalTracer().StartSpan("/toasts")
+	} else {
+		span = opentracing.GlobalTracer().StartSpan("/toasts", opentracing.ChildOf(context))
```

Comment:
Actually this could be simplified to `span = opentracing.GlobalTracer().StartSpan("/toasts", opentracing.ChildOf(context))` there is no need to check if the context is set.